### ISSUE TITLE
Is MSAL for Android using FIPS validated cryptography?

### ISSUE THREAD

**Issue by mdivkovic-speech** [AzureAD/microsoft-authentication-library-for-android#1602]
[FIPS](https://en.wikipedia.org/wiki/FIPS_140-2) Is a security standard that federal agencies in the US have to adhere to, and that means that the software they use also has to be built on top of FIPS validated cryptography.

Following this, could you answer these questions:

Is the MSAL Android client library using [FIPS approved cryptographic algorithms](https://cryptosense.com/blog/which-algorithms-are-fips-140-3-approved)?
Does MSAL use any proprietary cryptographic modules or libraries, or does it use the underlying runtime default cryptography?

**Comment by mdivkovic-speech** [comment]
Any news on this topic? Thank you in advance! Basically, I would like to know, if:

- You use only the FIPS approved algorithms and key sizes
- You use only the OS standard cryptography (Android Jetpack Security, Cipher, etc.), and don't have your own build of something like OpenSSL or similar bundled with the SDK to do the encryption or hashing with instead

**Comment by mdivkovic-speech** [comment]
OK, I've gone through the code myself, and I can see only system default cryptography being used. So in theory, if the crypto module has been FIPS validated for a given OEM hardware and OS combination, this should be fine.

The algorithms and key sizes also look OK. I noticed a single use of `MessageDigest.getInstance("SHA");`, but that's used in the Inspector app, and not in the SDK itself, so it doesn't matter for consumers of the SDK.

**Disclaimer: I am not a FIPS auditor, take everything I wrote here with a huge grain of salt and do your own research.**
